week2/project/src/App.css

```
@@ -0,0 +1,24 @@
+.product-list {
+  display: grid;
+  gap: 20px;
+  grid-auto-flow: row; /* This allows items to flow into a new row when there's not enough space in the column */
+}
+
+/* Responsive styles */
+@media (min-width: 576px) {
+  .product-list {
+    grid-template-columns: repeat(2, 1fr);
+  }
+}
+
+@media (min-width: 768px) {
+  .product-list {
+    grid-template-columns: repeat(3, 1fr);
+  }
+}
+
+@media (min-width: 992px) {
+  .product-list {
+    grid-template-columns: repeat(4, 1fr);
+  }
+}
```

The columns are looking good now! 🙌 But there is actually a way to do this all in one line with `grid-template-columns` and you don't have to use the media queries 🙂 Also, for the media queries you don't have to add `.product-list` all the time, you also do the media queries inside the `.product-list` class.

```suggestion
.product-list {
  display: grid;
  gap: 20px;
  grid-template-columns: repeat(auto-fill,minmax(300px,1fr));
}
```